Kirkstone migration and updates

HaberkornJonas opened the pull request:
### Changes Overview

- **[Migrating to kirkstone and made build pass.](https://github.com/axians/meta-microservicebus/commit/b947e71d155de11413ba13981de8df91b8135964)** (b947e71d155de11413ba13981de8df91b8135964)
  - Adapted the syntax (released in [Yocto 3.4](https://docs.yoctoproject.org/dev/migration-guides/release-3.4.html))
  - Edited LAYERSERIES_COMPAT to be kirkstone
  - Fixed QA_EMPTY_DIRS issue as /run wasn't empty due to msb-dam
  - DIRFILES disowning parent dir of files added/modified by msb-node-user
  - Recreated serialport recipe with devtool and updated to v10.4.0
  - Recreated dbus recipe with devtool

- **[Merging newer changes from default warrior branch into kirkstone.](https://github.com/axians/meta-microservicebus/commit/d640e2e7952e83e36b36490af165be817a86028b)** (d640e2e7952e83e36b36490af165be817a86028b)
  - Updated user password handling (see: https://github.com/axians/meta-microservicebus/commit/2ee85c24ca325f3096167b4dbc0acd009f6cd5fa)
  - Added support for NODE_EXTRA_CA_CERT (see: https://github.com/axians/meta-microservicebus/commit/042778bdb50236f2d8420341d2144b2838ade0fd)
  - Migrated msb-node-user 2.1.0 to support kirkstone
  - Resolved warning "microservicebus-node-service: recipe doesn't inherit features_check"
  - Resolved warning "the license listed Unknown was not in the licenses collected for recipe msb-node"

- **[Added recipe for msb-node@3.4.5](https://github.com/axians/meta-microservicebus/commit/1149c10556eb3d036e826418bf883b0d95d6ed3f)** (1149c10556eb3d036e826418bf883b0d95d6ed3f)

- **[Added inherit features_check to fix warning during build of npm-dam-scheduled-service](https://github.com/axians/meta-microservicebus/commit/57efc48b5c52ec4a918f321dc5f558bceeac3f80)** (57efc48b5c52ec4a918f321dc5f558bceeac3f80)

- **[Added recipe for nodejs v14 as v16 provided by meta-oe isn't supported by msb-node yet](https://github.com/axians/meta-microservicebus/commit/ddc54533a1a8439a430cb5c1a04f1ab8a3936047) (ddc54533a1a8439a430cb5c1a04f1ab8a3936047)**
  - Can be removed once this is fixed: https://github.com/axians/microservicebus-core/issues/91

wmmihaa committed:
updated user password handling

jonaskgandersson committed:
Add support for NODE_EXTRA_CA_CERT to mSB-node service

This adds the posibility to point to custom CA, for example if mSB
is running behind a proxy.

wmmihaa committed:
Merge pull request #47 from axians/feature/custom-ca

Add support for NODE_EXTRA_CA_CERT to mSB-node service

jonaskgandersson committed:
mSB-User cleanup old code, set default superuser name to msbsu to align with sudoers

jonaskgandersson committed:
mSB-node update license checksums

jonaskgandersson committed:
Remove old recipe version not used or maintained

HaberkornJonas committed:
Migrating to kirkstone and made build pass.

- Adapted the syntax (released in 3.4)
- Edited LAYERSERIES_COMPAT to be kirkstone
- Fixed QA_EMPTY_DIRS issue as /run wasn't empty due to msb-dam
- DIRFILES disowning parent dir of files added/modified by msb-node-user
- Recreated serialport recipe with devtool and updated to v10.4.0
- Recreated dbus recipe with devtool

HaberkornJonas committed:
Merging newer changes from default warrior branch into kirkstone.

- Updated user password handling (see: 2ee85c24ca325f3096167b4dbc0acd009f6cd5fa)
- Added support for NODE_EXTRA_CA_CERT (see: 042778bdb50236f2d8420341d2144b2838ade0fd)
- Migrated msb-node-user 2.1.0 to support kirkstone
- Resolved warning "microservicebus-node-service: recipe doesn't inherit features_check"
- Resolved warning "the license listed Unknown was not in the licenses collected for recipe msb-node"

HaberkornJonas committed:
Added recipe for msb-node@3.4.5

HaberkornJonas committed:
Added inherit features_check to fix warning during build of npm-dam-scheduled-service

HaberkornJonas committed:
Added recipe for nodejs v14 as v16 provided by meta-oe isn't supported by msb-node yet

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/nodejs/nodejs_14.18.1.bb:
```diff
@@ -0,0 +1,200 @@
+DESCRIPTION = "nodeJS Evented I/O for V8 JavaScript"
```
Better to use nodejs 16.14.2 as is delivered in Kirkstone and solve the issue in msb-core instead.
msb-core issue [91](https://github.com/axians/microservicebus-core/issues/91)

HaberkornJonas replied:
I agree.

Should I remove this nodejs v14 recipe and `PREFERRED_VERSION:nodejs ?= "14.18.1"` from layer.conf or do you want to keep it until https://github.com/axians/microservicebus-core/issues/91 is implemented?

I added it for convenience but can also move it to our app layer if you prefer to not have it here.

wmmihaa commented on meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb:
```diff
@@ -23,16 +23,18 @@ MSB_NODE_USER ?= "msb"
     sed -i -e 's:@MSB_NODE_GROUP@:${MSB_NODE_GROUP}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_NODE_HOST@:${MSB_NODE_HOST}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_DAM_SOCKETPATH@:${MSB_DAM_SOCKETPATH}:g' ${WORKDIR}/microservicebus-node.service
+    sed -i -e 's:@MSB_NODE_EXTRA_CA_CERTS@:${MSB_NODE_EXTRA_CA_CERTS}:g' ${WORKDIR}/microservicebus-node.service
```
Can you describe how this is used in local.conf

HaberkornJonas replied:
We use it like this in local.conf: 
`MSB_NODE_EXTRA_CA_CERTS = "/usr/local/share/ca-certificates/zscaler-root-ca.crt"`

It's currently the certificate from the [meta-vinci-energies](https://github.com/axians/meta-vinci-energies) layer but we might need to add other certificates at some point and would change the name of the file in that case.

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb:
```diff
@@ -23,16 +23,18 @@ MSB_NODE_USER ?= "msb"
 MSB_NODE_GROUP ?= "msb"
 MSB_NODE_HOST ?= "microservicebus.com"
 MSB_DAM_SOCKETPATH ?= "/var/run/dam"
+MSB_NODE_EXTRA_CA_CERTS ?= ""
```
In local.conf set MSB_NODE_EXTRA_CA_CERTS = "PATH-TO-CA.crt"

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb:
```diff
@@ -33,7 +33,7 @@ inherit npm
 
```
Remove node-pty recipe, msb-core fetch this on demand.

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb:
```diff
@@ -33,7 +33,7 @@ inherit npm
 
```
Remove node-pty recipe, msb-core fetch this on demand.

HaberkornJonas replied:
Removed the **nodejs-pty** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/16ef03fa627f06aa6b8dd4353f0702cdbc87acd5

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb:
```diff
@@ -1,78 +1,80 @@
+# Recipe created by recipetool
```
Remove nodejs-dbus recipe, msb-core fetch this on demand.

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb:
```diff
@@ -1,78 +1,80 @@
+# Recipe created by recipetool
```
Remove nodejs-dbus recipe, msb-core fetch this on demand.

HaberkornJonas replied:
* Removed the **nodejs-dbus** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/fdad0fea0aeab1970fff0462c9923a79b2d68b34
* Edited **README** to remove occurrences of nodejs-dbus

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb:
```diff
@@ -0,0 +1,75 @@
+# Recipe created by recipetool
```
Remove node-serialport recipe, msb-core fetch this on demand.

HaberkornJonas replied:
* Removed the **node-serialport** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/ec66b71e75e941b89f45fdb6c8329761356469eb
* Removed **packagegroup-microservicebus-extra** in https://github.com/axians/meta-microservicebus/pull/49/commits/7cbc4c41d07079808d5eedc865e355852d813e7c as its only dependency (node-serialport) was removed
* Edited **README** to remove occurrences of node-serialport and microservicebus-extra

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb:
```diff
@@ -0,0 +1,75 @@
+# Recipe created by recipetool
```
Remove node-serialport recipe, msb-core fetch this on demand.

HaberkornJonas replied:
* Removed the **node-serialport** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/ec66b71e75e941b89f45fdb6c8329761356469eb
* Removed **packagegroup-microservicebus-extra** in https://github.com/axians/meta-microservicebus/pull/49/commits/7cbc4c41d07079808d5eedc865e355852d813e7c as its only dependency (node-serialport) was removed
* Edited **README** to remove occurrences of node-serialport and microservicebus-extra

jonaskgandersson commented on meta-microservicebus-node/conf/layer.conf:
```diff
@@ -11,9 +11,12 @@ BBFILE_PRIORITY_meta-microservicebus-node= "16"
 SPDXLICENSEMAP[WTFPLv1] = "WTFPL"
+
+# Using NodeJS v14 until v16+ is supported
+PREFERRED_VERSION:nodejs ?= "14.18.1"
```
@wmmihaa is fixing nodejs v16 support.

jonaskgandersson replied:
@wmmihaa is fixing nodejs v16 support.

jonaskgandersson commented on meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node/microservicebus-node_3.4.3.bb:
```diff
@@ -259,107 +259,107 @@ S = "${WORKDIR}/npm"
 
```
Will get new msb-node version for nodejs V16

jonaskgandersson replied:
Will get new msb-node version for nodejs V16

jonaskgandersson reviewed (commented):
We will add support for node v16 instead of changing version in Yocto.

We can remove recipes:

- node-pty
- nodejs-dbus
- nodejs-serialport

As we never install them in image, msb-core installs them on demand.

wmmihaa committed:
Merge pull request #48 from axians/cleanup

Cleanup old versions

HaberkornJonas committed:
Removing nodejs-serialport recipe
As requested in https://github.com/axians/meta-microservicebus/pull/49#discussion_r892465201

HaberkornJonas committed:
Removing packagegroup msb-extra
As its dependences are now empty.

HaberkornJonas committed:
Removing nodejs-dbus recipe
As requested in https://github.com/axians/meta-microservicebus/pull/49#discussion_r892464791

HaberkornJonas committed:
Removing node-pty recipe
As requested in https://github.com/axians/meta-microservicebus/pull/49#discussion_r892464105

HaberkornJonas committed:
Merging last PR from warrior branch
See PR #48 - https://github.com/axians/meta-microservicebus/pull/48

HaberkornJonas committed:
Adding inherit features_check
To resolve "doesn't inherit features_check" warning

HaberkornJonas commented:
@wmmihaa, @jonaskgandersson - I saw that you merged #48 into the warrior branch, as it's the default branch I also merged it here to be up-to-date with the latest changes (8e54b2bc0c08d547cf1154002cea566d865e4a47) and fixed a features_check warning (84958f1729f6833d081f245262fc037da8a53fa3)

HaberkornJonas committed:
Renaming msb-node shrinkwrap file

HaberkornJonas committed:
Deleted msb-node v3.4.3 and added v4.0.0

HaberkornJonas committed:
Fixed typo + added nodejs-native preferred version

HaberkornJonas committed:
Added NodeJS v14 info in readme

HaberkornJonas committed:
Added README file extensions

HaberkornJonas committed:
Improved readme render on GitHub
